deluge/common.py

```diff
@@ -44,6 +45,14 @@
 
     os.environ['SSL_CERT_FILE'] = where()
 
+# On windows sock.if_nameindex() is only supported on 3.8+ and does not return a uuid.
+# Try ifaddr to verify network interfaces.
+if platform.system() in ('Windows', 'Microsoft'):
```

Comment:
later on we do linux_check() or osx_check() to not run ifaddr on linux or mac as it should not be needed since those platforms can use socket.if_nameindex().